File: backend/src/main/java/access_control/entity/Space.java
package access_control.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Entity
@Table(name = "Space")
public class Space {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 100)
    private String name;

    @Column
    private int capacity;

    @Column(nullable = false, length = 500)
    private String description;

    @Column(nullable = false, length = 30)
    private String status;

}

File: frontend/lib/models/space.dart
class Space {
  final int id;
  final String name;
  final int capacity;
  final String description;
  final String status;

  Space({
    required this.id,
    required this.name,
    required this.capacity,
    required this.description,
    required this.status,
  });

  factory Space.fromJson(Map<String, dynamic> json) {
    return Space(
      id: json['id'],
      name: json['name'],
      capacity: json['capacity'],
      description: json['description'],
      status: json['status'],
    );
  }
}